### PR TITLE
Fix #1976 name of the user is visible in navigationView

### LINKED ISSUE #1976: Name of the user should be visible in the navigationView
**Actual Behaviour**

Currently if the user has logged in,the email id of the user is displayed in the navigationView

**Expected Behaviour**

If the user is providing his/her name,the name of the user should be visible in the navigationView instead of his email

**Steps to reproduce it**

Sign up and change your name in the EditProfileActivity.
Observe the navigationView

**Screenshots of the issue**
screenshot_20171104-132817

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by nikit19**
Fixes #1976 

Changes:If the user logs in and changes his/her name in the editProfileActivity, that name will be displayed in the navigationView instead of his email

Screenshots for the change: 
screenshot_20171109-222151

### CODE REVIEW

**Review by scandindian — CHANGES_REQUESTED** [review]
When you are using NAME as two parts, use FIRST_NAME and LAST_NAME. Similarly for variable names.

**Comment by nikit19** [comment]
@heysadboy done with the requested changes

**Review by scandindian — APPROVED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 1]
```diff
@@ -242,6 +258,7 @@ protected void onPause() {
     protected void onResume() {
         super.onResume();
         OpenEventApp.getEventBus().register(this);
+       setUserProfileMenuItem();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: The indentation is off here

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 2]
```diff
@@ -227,6 +227,22 @@ public void onServiceDisconnected(ComponentName name) {
...
+        MenuItem userProfileMenuItem = navigationView.getMenu().findItem(R.id.nav_user_profile);
+        if (AuthUtil.isUserLoggedIn()) {
+            String email = SharedPreferencesUtil.getString(ConstantStrings.USER_EMAIL, null);
+            String firstName = SharedPreferencesUtil.getString(ConstantStrings.USER_FIRST_NAME, null);
+            String lastName = SharedPreferencesUtil.getString(ConstantStrings.USER_LAST_NAME, null);
+
+            if (firstName != null && lastName != null && !firstName.isEmpty() && !lastName.isEmpty())
+                userProfileMenuItem.setTitle(firstName + " " + lastName);
+            else if (firstName != null && !firstName.isEmpty())
+                userProfileMenuItem.setTitle(firstName);
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: I think, you can use `TextUtils.isEmpty` for firstName and then just set text as `firstName + lastName`

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by iamareebjamal** [comment]
@nikit19 Please resolve conflicts

**Comment by nikit19** [comment]
@iamareebjamal done with the requested changes

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 3]
```diff
@@ -227,6 +228,22 @@ public void onServiceDisconnected(ComponentName name) {
...
     }
 
+    private void setUserProfileMenuItem(){
+        MenuItem userProfileMenuItem = navigationView.getMenu().findItem(R.id.nav_user_profile);
+        if (AuthUtil.isUserLoggedIn()) {
+            String email = SharedPreferencesUtil.getString(ConstantStrings.USER_EMAIL, null);
+            String firstName = SharedPreferencesUtil.getString(ConstantStrings.USER_FIRST_NAME, null);
+            String lastName = SharedPreferencesUtil.getString(ConstantStrings.USER_LAST_NAME, null);
+
+            if (firstName != null && lastName != null && !TextUtils.isEmpty(firstName) && !TextUtils.isEmpty(lastName))
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: TextUtils.isEmpty also checks for null, no need to do it explicitly. Also, only check for isEmpty(firstName) and is it is not empty, show firstName + " " + lastName

Showing only firstName if lastName is null is redundant as `X + " " + null` is `X`

Use isEmpty for email check as well

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 3 above)
Comment (nikit19) [reply android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: done with the requested changes

**Review by iamareebjamal — DISMISSED** [review]

**Comment by iamareebjamal** [comment]
@nikit19 Please resolve conflicts

**Comment by nikit19** [comment]
conflicts are resolved

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**